Source/AviationWeather.Test/Source/metar_parser_tests.cpp

```diff
@@ -0,0 +1,447 @@
+    Assert::AreEqual(std::string("KSFO"), metar.station_identifier);
+
+    // EBGE
+    std::string m2("EGBE ");
+    metar::parsers::parse_station_identifier(metar, m2);
+    Assert::AreEqual(std::string("EGBE"), metar.station_identifier);
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_ObservationTime()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(uint8_t(1), metar.report_time.day_of_month);
+    Assert::AreEqual(uint8_t(0), metar.report_time.hour_of_day);
+    Assert::AreEqual(uint8_t(0), metar.report_time.minute_of_hour);
+
+    // 8th, 17:53
+    std::string m1("081753Z ");
+    metar::parsers::parse_observation_time(metar, m1);
+    Assert::AreEqual(uint8_t(8), metar.report_time.day_of_month);
+    Assert::AreEqual(uint8_t(17), metar.report_time.hour_of_day);
+    Assert::AreEqual(uint8_t(53), metar.report_time.minute_of_hour);
+
+    // 20th, 03:28
+    std::string m2("200328Z ");
+    metar::parsers::parse_observation_time(metar, m2);
+    Assert::AreEqual(uint8_t(20), metar.report_time.day_of_month);
+    Assert::AreEqual(uint8_t(3), metar.report_time.hour_of_day);
+    Assert::AreEqual(uint8_t(28), metar.report_time.minute_of_hour);
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_Modifier()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(metar::modifier_type::automatic, metar.modifier);
+
+    // AUTO
+    std::string m1("AUTO ");
+    metar::parsers::parse_modifier(metar, m1);
+    Assert::AreEqual(metar::modifier_type::automatic, metar.modifier);
+
+    // COR
+    std::string m2("COR ");
+    metar::parsers::parse_modifier(metar, m2);
+    Assert::AreEqual(metar::modifier_type::corrected, metar.modifier);
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_Wind()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(UINT16_MAX, metar.wind_group.direction);
+    Assert::AreEqual(uint8_t(0), metar.wind_group.wind_speed);
+    Assert::AreEqual(uint8_t(0), metar.wind_group.gust_speed);
+    Assert::AreEqual(UINT16_MAX, metar.wind_group.variation_lower);
+    Assert::AreEqual(UINT16_MAX, metar.wind_group.variation_upper);
+    Assert::AreEqual(speed_unit::kt, metar.wind_group.unit);
+
+    // 240 @15kts
+    std::string m1("24015KT ");
+    metar::parsers::parse_wind(metar, m1);
+    Assert::AreEqual(uint16_t(240), metar.wind_group.direction);
+    Assert::AreEqual(uint8_t(15), metar.wind_group.wind_speed);
+    Assert::AreEqual(uint8_t(0), metar.wind_group.gust_speed);
+    Assert::AreEqual(UINT16_MAX, metar.wind_group.variation_lower);
+    Assert::AreEqual(UINT16_MAX, metar.wind_group.variation_upper);
+    Assert::AreEqual(speed_unit::kt, metar.wind_group.unit);
+
+    // 120 @ 2kts, variable between 100 and 140
+    std::string m2("12002KT 100V140 ");
```

Comment:
Can we change this to a valid METAR?  VRB is used when wind speed is 6 knots and less; upper and lower variation limits are only used when greater than 6 knots.  Perhaps split it into two tests, VRB02KT and 12010KT 100V140?